rfc/group-by-space-provider.md

```diff
@@ -0,0 +1,58 @@
+# Group by provider / sign by tenant
+
+by: @vasco-santos
+
+ https://filecoinproject.slack.com/archives/C02BZPRS9HP/p1689950046118329?thread_ts=1689873032.504899&cid=C02BZPRS9HP
+
+building a bit on top of this, and thinking about the flow of producers send us pieces, looks like this would fit as part of the w3-aggregation spec as a UCAN capability.
+I think later on (after migration) it can be done as an effect of `store/add` together with `piece/build` (or similar).
```

Comment:
I think capturing that in the effect is a cool ideal, as it will appear in the receipt and would allow recipient to look it up.

We should also consider how clients could submit `piece` CIDs to us. Perhaps we could have `piece/add` that users could invoke and have `piece/verify` instead of `piece/build` that way if we already have a car and piece we could avoid that effect.

In the future we could also use `piece/add` in place of S3 event to kick things off, and if user never did `piece/add` then it does not ends up in filecoin and that on them.

Reply:
ok, so `piece/add` followed by `piece/verify` that would trigger the pipeline if verified. Still need to think more on how we would skip verify in some steps. Maybe invocation is just triggered and the handler can decide to skip and state reasoning in receipt